src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt

```diff
@@ -12,6 +16,71 @@ abstract class MavenPublishBaseExtension(
   private val project: Project
 ) {
 
+  private var nexusOptions: NexusOptions? = null
+
+  /**
+   * Sets up Maven Central publishing through Sonatype OSSRH by configuring the target repository. Gradle will then
+   * automatically create a `publishAllPublicationsToMavenRepository` task as well as include it in the general
+   * `publish` task. If the current version ends with `-SNAPSHOT` the artifacts will be published to Sonatype's snapshot
+   * repository instead.
+   *
+   * This expects you provide your Sonatype user name and password through Gradle properties called
+   * `mavenCentralRepositoryUsername` and `mavenCentralRepositoryPassword`.
+   *
+   * The `closeAndReleaseRepository` task is automatically configured for Sonatype OSSRH using the same credentials.
+   *
+   * @param host the instance of Sonatype OSSRH to use
+   * @param stagingRepositoryId optional parameter to upload to a specific already created staging repository
+   */
+  @Incubating
+  fun publishToMavenCentral(host: SonatypeHost, stagingRepositoryId: String? = null) {
+    project.gradlePublishing.repositories.maven { repo ->
+      repo.name = "mavenCentral"
+      if (stagingRepositoryId != null) {
+        repo.setUrl("${host.rootUrl}/service/local/staging/deployByRepositoryId/$stagingRepositoryId/")
+      } else {
+        repo.setUrl("${host.rootUrl}/service/local/staging/deploy/maven2/")
+      }
+      repo.credentials(PasswordCredentials::class.java)
+
+      project.afterEvaluate {
+        if (it.version.toString().endsWith("SNAPSHOT")) {
+          if (stagingRepositoryId != null) {
+            throw IllegalArgumentException("Staging repositories are not supported for SNAPSHOT versions.")
+          }
+          repo.setUrl("${host.rootUrl}/content/repositories/snapshots/")
+        }
+      }
+    }
+
+    nexusOptions {
+      it.baseUrl = "${host.rootUrl}/service/local/"
+      it.repositoryUsername = project.findOptionalProperty("mavenCentralRepositoryUsername")
+      it.repositoryPassword = project.findOptionalProperty("mavenCentralRepositoryPassword")
+    }
+  }
+
+  internal fun nexusOptions(action: Action<NexusOptions>) {
+    var nexusOptions = this.nexusOptions
+    if (nexusOptions == null) {
+      nexusOptions = NexusOptions(null, null, null, null)
+      this.nexusOptions = nexusOptions
+
+      @Suppress("SwallowedException")
+      try {
+        project.rootProject.tasks.named("closeAndReleaseRepository")
+      } catch (e: UnknownTaskException) {
+        project.rootProject.tasks.register("closeAndReleaseRepository", CloseAndReleaseRepositoryTask::class.java) {
```

Comment:
I would just make the task regardless, worst case it's unused. Tasks should be free

Reply:
aba6546

Reply:
Reverted in df3ad06 because of a misunderstanding. Is it ok if I replace this part in a separate PR, because we used the same approach in the old code as well?

Reply:
👍